operations/helm/charts/mimir-distributed/templates/ingester/ingester-statefulset.yaml

```
@@ -34,13 +34,21 @@ spec:
           {{- toYaml .Values.ingester.persistentVolume.annotations | nindent 10 }}
         {{- end }}
       spec:
+        {{- if $rolloutZone.storageClass}}
+        {{- if (eq "-" $rolloutZone.storageClass) }}
+        storageClassName: {{ .Values.ingester.persistentVolume.storageClass }}
+        {{- else }}
+        storageClassName: {{ $rolloutZone.storageClass }}
+        {{- end }}
+        {{- else }}
         {{- if .Values.ingester.persistentVolume.storageClass }}
         {{- if (eq "-" .Values.ingester.persistentVolume.storageClass) }}
         storageClassName: ""
         {{- else }}
         storageClassName: "{{ .Values.ingester.persistentVolume.storageClass }}"
         {{- end }}
         {{- end }}
+        {{- end }}
```

Same as my comment on the alertmanager template

```suggestion
        {{- $storageClass := default .Values.ingester.persistentVolume.storageClass $rolloutZone.storageClass }}
        {{- if $storageClass }}
        {{- if (eq "-" $storageClass) }}
        storageClassName: ""
        {{- else }}
        storageClassName: {{ $storageClass }}
        {{- end }}
        {{- end }}
```